clang/include/clang/Tooling/DependencyScanning/DependencyScanningWorker.h

```diff
@@ -113,6 +118,9 @@ class DependencyScanningWorker {
   /// worker. If null, the file manager will not be reused.
   llvm::IntrusiveRefCntPtr<FileManager> Files;
   ScanningOutputFormat Format;
+
+  const char *ModuleName = nullptr;
+  std::unique_ptr<llvm::MemoryBuffer> FakeMemBuffer;
```

Comment:
Sorry I didn't catch this earlier.

Looking at the dependency scanner architecture again, I don't think these fields should live inside `DependencyScanningWorker`. They introduce a state that cannot be shared between multiple invocations of the worker. If you call `Worker->setModuleName()`, you'd need to remember to call `Worker->unsetModuleName()` to clear the temporary state.

The only place the module name is being used is in `DependencyScanningWorker::computeDependencies` where we're setting up `DependencyScanningAction`. Why don't we pass the optional module name as a parameter to the function?

The same goes for `FakeMemBuffer`. This can be just a variable inside `computeDependencies`, can't it?

Reply:
Do we want to pass the module name as an optional argument to all the functions that indirectly call `computeDependencies`, e.g., `computeDependenciesForClangInvocation`, `getFullDependencies`, `getFileDependencies`?

`DependencyScanningTool` has member `ModuleName` too. Is it necessary to remove that too?

Reply:
Yes, the `DependencyScanningTool` instances are being reused for multiple dependency scans, so it's the same deal as `DependencyScanningWorker`.